### PR TITLE
Fix guide user meta

### PR DESCRIPTION
**Pull request by atsanna**
The Users And Security-> user_meta guide is not displayed.

I tried to fix the problem.

In the visualization file I have made sure that the subfolders are separated with the "-"

At this point I managed it in the controller.

After these changes, the Guidecollection library gave me an error in the loop for calculating the previous and next pages.

Not sure if I did a good job, but it seems to work (I have created several folders and subfolders) ...

### CODE REVIEW

**Comment by atsanna** [comment]
I think I fixed the error (due to my changes) in the GuideCollection library when calculating the previous and next pages: the subfolders had to be removed from the count ...

The view code, after my modifications, was becoming too complex: I thought of cleaning up by adding a helper function for recursive search of subfolders

File: bonfire/Modules/Guides/Libraries/GuideCollection.php [hunk 1]
```diff
@@ -147,9 +147,21 @@ public function pageLinks()
...
+				$pages ,
+				function($k) {
+					//If the key is a string, it is a subfolder
+					if(!is_numeric($k)) {
+						return $k;
+					}
+				},
+				ARRAY_FILTER_USE_KEY)
+		);
+
```
Comment (atsanna) [inline bonfire/Modules/Guides/Libraries/GuideCollection.php:161]: I think I fixed the error in the GuideCollection library when calculating the previous and next pages: the subfolders had to be removed from the count ...

File: bonfire/Modules/Guides/Helpers/toc_helper.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	{
+		// Name of folder
+		$string = '<h3>' . esc(ucwords(trim(Bonfire\Modules\Guides\Libraries\GuideCollection::formatPage($folder), ' /'))) . '</h3>
+					<ul class="list-unstyled px-4">';
+
+		 foreach($page as $sub_folder => $row) {
+			 if( is_numeric($sub_folder) ) {
+				 // Is file
+				 $string .= '<li>
+								<a href="/' . ADMIN_AREA . '/guides/' . $alias . '-' . trim($folder, ' /')  . '/' . trim($row, ' /') .'">
```
Comment (atsanna) [inline bonfire/Modules/Guides/Helpers/toc_helper.php:19]: In order not to make changes to the routes, I thought of separating the folder tree with the "-" character and passing the whole segment to the controller

File: bonfire/Modules/Guides/Libraries/GuideCollection.php [hunk 3]
```diff
@@ -147,9 +147,21 @@ public function pageLinks()
...
         $previous = null;
         $next = null;
-        for($i=0; $i < count($pages); $i++) {
+        for($i=0; $i < count($pages) - $count_sub_folders; $i++) {
```
Comment (atsanna) [inline bonfire/Modules/Guides/Libraries/GuideCollection.php:164]: I think I corrected the error (due to my changes) in the GuideCollection library when calculating the previous and next pages: the subfolders needed to be removed from the count ...

Comment (lonnieezell) [reply bonfire/Modules/Guides/Libraries/GuideCollection.php:164]: Thanks! Hadn't gotten back to cleaning that up yet.

**Comment by lonnieezell** [comment]
So it didn't resolve the next/previous links issue. And I'm not sure I'm loving the sub-folder solution but it is working for now. I might revisit it later when I get some more docs updated and organized. We'll see how it works out. 

Thanks!

**Pull request merged**